### DISCUSSION TITLE
技巧分享：Quicker一步打开本地PDF（在沙拉查词PDF阅读器独立窗口）

### DISCUSSION THREAD

**Discussion by hqyyqh** [crimx/ext-saladict#1005] [Show and tell]
# 需求

如果仅是阅读英文PDF文献，沙拉查词内置的PDF阅读器可能已经足够，但是打开本地pdf文件的操作比较繁琐。

如果直接在资源管理器中，**一步到位**在独立窗口中打开PDF，那就很方便了。

> Edge 中暂无法使用。

# 效果

借助 Quicker（没错，又是它）。

![saladict_pdf_2](https://user-images.githubusercontent.com/38676455/92996954-1a882c80-f542-11ea-88c6-5db3bdeb96c8.gif)

# 配置

1. 沙拉查词开启【允许访问文件网址】
2. 安装Quicker以及Quicker动作【[沙拉查词PDF](https://getquicker.net/sharedaction?code=2d74bec5-4a7d-4241-d793-08d856359351)】

![saladict_pdf_tips](https://user-images.githubusercontent.com/38676455/92996612-7d2bf900-f53f-11ea-8306-a1e5ec716e8a.png)

# 实现

获取文件路径后，直接用 app 模式`--app=URL`打开与`chrome-extension://cdonnmffkdaoajfknoeeecmchibpmkmg/assets/pdf/web/viewer.html`组合后的路径。

例如，对于本地文件 `D:/test/test.pdf`，按下 WIN+R，在输入框中输入回车 `chrome --app=chrome-extension://cdonnmffkdaoajfknoeeecmchibpmkmg/assets/pdf/web/viewer.html?file=file:///D:/test/test.pdf` 即可

**Comment by ning604** [comment]
![image](https://user-images.githubusercontent.com/71740131/98040232-ec122980-1ded-11eb-901e-c816cfd16f0a.png)
请问我为什么打开后是这个样子的？

**Comment by hqyyqh** [comment]
> 请问我为什么打开后是这个样子的？

需要按照截图上的指引，开启【沙拉查词】的【允许访问文件网址】权限

**Comment by ning604** [comment]
> > 请问我为什么打开后是这个样子的？
> 
> 需要按照截图上的指引，开启【沙拉查词】的【允许访问文件网址】权限

我按照指示做过了，第一次的时候成功了的，昨天突然打不开，说是无法用chromeApp 模式打开网址失败，请确保安装chrome并安装到环境变量path中，我安装了也把环境变量配置好了，就是c盘的那个路径，文本截图翻译的那个能正常工作，但是这个不可以

**Comment by hqyyqh** [comment]
> 我按照指示做过了，第一次的时候成功了的，昨天突然打不开，说是无法用 chromeApp 模式打开网址失败，请确保安装 chrome 并安装到环境变量 path 中，我安装了也把环境变量配置好了，就是 c 盘的那个路径，文本截图翻译的那个能正常工作，但是这个不可以

按下 Win+R，输入 chrome 回车，是否可以打开浏览器？或者是输入 `chrome --app=https://baidu.com --start-maximized` 回车，是否可以打开一个独立窗口？

不行的话，可以试下重新覆盖安装chrome。

**Comment by ning604** [comment]
> > 我按照指示做过了，第一次的时候成功了的，昨天突然打不开，说是无法用 chromeApp 模式打开网址失败，请确保安装 chrome 并安装到环境变量 path 中，我安装了也把环境变量配置好了，就是 c 盘的那个路径，文本截图翻译的那个能正常工作，但是这个不可以
> 
> 按下 Win+R，输入 chrome 回车，是否可以打开网址？或者是输入 `chrome --app=https://baidu.com --start-maximized` 回车，是否可以打开一个独立窗口？
> 
> 不行的话，可以试下重新覆盖安装chrome。

好的，我重新安一下，谢谢
